.github/workflows/deploy_dev_stage_prod.yml

```diff
@@ -0,0 +1,128 @@
+name: Deploy to Dev, Stage, and Prod
+
+on:
+  workflow_call:
+    inputs:
+      commit-identifier:
+        description: SHA or tag to apply, (optional) when omitted, the head of the selected branch is used
+        type: string
+        required: true
+      terraform-workspace:
+        description: Terraform workspace that is used
+        type: string
+        required: true
+      terraform-root: 
+        description: Root directory of terraform, default to 'terraform/${ inputs.environment }'
+        type: string
+        required: false
```

Comment:
terraform workspace and terraform root are going to be troublesome inputs since the value may be different per environment and this workflow will deploy to all environment separately. But you'll probably have to figure that out when you try to wire it up with a real repo...